### PR TITLE
Python 3 support

### PR DESCRIPTION
**Pull request by mherrmann**

### CODE REVIEW

File: signalr/transports/_transport.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from abc import abstractmethod
 import json
-import urllib
+import urllib.request, urllib.parse, urllib.error
```
Comment (u-abramchuk) [inline signalr/transports/_transport.py]: Looks like it breaks Python 2.X compatibility. https://docs.python.org/2.7/library/urllib.html

File: README.md [hunk 2]
```diff
@@ -15,7 +15,9 @@ In Debian based distributions (such as Ubuntu and Raspbian) they are called `pyt
 
 ## Compatibility
 
-Only compatible with Python 2, not Python 3.
+Compatible with Python 2 and 3.
+
+For Python 3, you need to install a version of `gevent` >= 1.1.
```
Comment (u-abramchuk) [inline README.md]: The latests version of gevent available is 1.1rc1. https://pypi.python.org/pypi/gevent#downloads

File: signalr/transports/_transport.py (hunk 1 above)
Comment (mherrmann) [inline signalr/transports/_transport.py]: You are absolutely right, sorry. I had forgotten a commit. It is now on the
branch for which I submitted the pull request. Please check again!

Thanks
Michael

On 10 December 2015 at 14:45, Uladzimir Abramchuk <notifications@github.com>
wrote:

> In signalr/transports/_transport.py
> <https://github.com/TargetProcess/signalr-client-py/pull/2#discussion_r47226764>
> :
>
> > @@ -1,6 +1,6 @@
> >  from abc import abstractmethod
> >  import json
> > -import urllib
> > +import urllib.request, urllib.parse, urllib.error
>
> Looks like it breaks Python 2.X compatibility.
> https://docs.python.org/2.7/library/urllib.html
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TargetProcess/signalr-client-py/pull/2/files#r47226764>
> .
>



-- 
[image: Inline image 1]

Michael Herrmann, MSc
Alser Straße 18/26
1090 Wien
Tel.: +43 699 11 65 16 40
Email: michael@herrmann.io

**Comment by u-abramchuk** [comment]
Also make sure to update setup.py and DESCRIPTION.rst please to reflect the changes.

**Comment by mherrmann** [comment]
I've updated README, setup.py and DESCRIPTION.rst. Is this what you had in mind?

**Comment by u-abramchuk** [comment]
Yes, exactly. Now it looks much better. 

Could you please resolve merge conflicts?

**Comment by mherrmann** [comment]
I've created a new commit which is based off the latest version of branch `develop` and includes all changes necessary for Python 3 compatibility. You should now be able to merge the PR without any conflicts :)

**Comment by u-abramchuk** [comment]
LGTM.

**Pull request merged**